### PR TITLE
Feat/taxi booking management

### PR DESCRIPTION
**Pull request by ganilala**

### CODE REVIEW

**Review by oltahasimja — DISMISSED** [review]

File: pentasharp/Controllers/TaxiBookingController.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+        public TaxiBookingController(AppDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("SearchAvailableTaxis")]
+        public IActionResult GetCompanies()
```
Comment (GentaSL) [inline pentasharp/Controllers/TaxiBookingController.cs]: move the business logic you are doing in the controllers to a service, create its interface, register the service in program.cs and inject it propely in the controller. Same as per the comment i made in Oltas PR.

File: pentasharp/Controllers/TaxiBookingController.cs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                return NotFound(new StandardApiResponse<string>
+                {
+                    Success = false,
+                    Message = ResponseMessages.NotFound,
+                    Code = ResponseCodes.NotFound
+                });
+            }
+
+            var viewModel = new TaxiBookingViewModel
+            {
```
Comment (GentaSL) [inline pentasharp/Controllers/TaxiBookingController.cs]: auto mappers!!

File: pentasharp/Controllers/TaxiBookingController.cs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                return BadRequest(new { success = false, message = "Invalid data." });
+
+            var booking = await _context.TaxiBookings.FirstOrDefaultAsync(b => b.BookingId == model.BookingId);
+            if (booking == null)
+                return NotFound(new { success = false, message = "Booking not found." });
+
+            booking.PickupLocation = model.PickupLocation;
+            booking.DropoffLocation = model.DropoffLocation;
+            booking.BookingTime = model.BookingTime;
+            if (Enum.TryParse(typeof(ReservationStatus), model.Status.ToString(), true, out var status))
```
Comment (GentaSL) [inline pentasharp/Controllers/TaxiBookingController.cs]: use braces, 
if(something) {
do something;
} else {
something else
}

File: pentasharp/Controllers/TaxiBookingController.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPut("UpdateBooking")]
+        public async Task<IActionResult> UpdateBooking([FromBody] EditTaxiBookingViewModel model)
+        {
+            if (!ModelState.IsValid)
+                return BadRequest(new { success = false, message = "Invalid data." });
+
+            var booking = await _context.TaxiBookings.FirstOrDefaultAsync(b => b.BookingId == model.BookingId);
+            if (booking == null)
```
Comment (GentaSL) [inline pentasharp/Controllers/TaxiBookingController.cs]: use braces,
if(something) {
do something;
}

File: pentasharp/ViewModel/TaxiBooking/ConfirmBookingViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace pentasharp.ViewModel.TaxiBooking
+{
+    /// <summary>
+    /// Represents the data required to create a taxi booking.
+    /// </summary>
+    public class CreateBookingViewModel
```
Comment (GentaSL) [inline pentasharp/ViewModel/TaxiBooking/ConfirmBookingViewModel.cs]: move this viewmodel and the others in this folder to a TaxiRequests folder you'll create under Models. I dont think ViewModel folder separation is necessary. I proposed something similar to Olta's PR https://github.com/starlabspro-internship/pentasharp/pull/24/files#r1867294136

File: pentasharp/Models/Entities/TaxiBookings.cs [hunk 6]
```diff
@@ -11,6 +11,9 @@ public class TaxiBookings
         [Key]
         public int BookingId { get; set; }
 
+        [Required]
```
Comment (GentaSL) [inline pentasharp/Models/Entities/TaxiBookings.cs]: remove data annotations as we agreed on removing them, add missing summaries for the whole model. rename this TaxiBooking model to the corresponding TaxiBookingRequest or TaxiBookingResponse and the folder shouldnt be named Entities

Comment (GentaSL) [reply pentasharp/Models/Entities/TaxiBookings.cs]: @ganilala this is not fixed yet

**Review by GentaSL — APPROVED** [review]

**Review by oltahasimja — APPROVED** [review]

**Pull request merged**